programs/marginfi/tests/user_actions/liquidate.rs

```diff
@@ -257,32 +277,78 @@ async fn marginfi_account_liquidation_success(
 
     let collateral_bank = collateral_bank_f.load().await;
     let debt_bank = debt_bank_f.load().await;
-
-    // Check liquidator collateral and debt balances
-    let expected_collateral_mint_liquidator_balance = I80F48::from(native!(
-        liquidator_deposit_amount + liquidate_amount,
+    let liquidate_amount_native = I80F48::from(native!(
+        liquidate_amount,
         collateral_bank_f.mint.mint.decimals,
         f64
     ));
 
+    // Check liquidator collateral and debt balances
+    let expected_collateral_mint_liquidator_balance = if liquidator_deposit_amount > 0.0 {
+        I80F48::from(native!(
+            liquidator_deposit_amount,
+            collateral_bank_f.mint.mint.decimals,
+            f64
+        ))
+        .checked_mul(collateral_bank.asset_share_value.into())
+        .unwrap()
+        .checked_add(liquidate_amount_native)
+        .unwrap()
+    } else {
+        liquidate_amount_native
+            .checked_sub(
+                I80F48::from(native!(
+                    -liquidator_deposit_amount,
+                    collateral_bank_f.mint.mint.decimals,
+                    f64
+                ))
+                .checked_mul(collateral_bank.liability_share_value.into())
+                .unwrap(),
+            )
+            .unwrap()
+    };
+
     let collateral_mint_liquidator_balance = collateral_bank.get_asset_amount(
         liquidator_mfi_ma.lending_account.balances[liquidator_collateral_index]
             .asset_shares
             .into(),
     )?;
-    assert_eq!(
+    assert_eq_noise!(
         expected_collateral_mint_liquidator_balance,
-        collateral_mint_liquidator_balance
+        collateral_mint_liquidator_balance,
+        1.
     );
 
     let debt_paid_out = liquidate_amount * 0.975 * collateral_bank_f.get_price().await
         / debt_bank_f.get_price().await;
+    let debt_paid_out_native =
+        I80F48::from(native!(debt_paid_out, debt_bank_f.mint.mint.decimals, f64));
+
+    // We have to account for updated share value here due to 1 second of delay we introduced to check last_update
```

Comment:
it might be better to validate the timing update in a separate test and keep the old one with precise assertions too. better to have both imo.